exonum-java-binding/core/rust/src/storage/entry.rs

```diff
@@ -25,7 +25,8 @@ use handle::{self, Handle};
 use storage::db::{Value, View, ViewRef};
 use utils;
 
-type Index<T> = Entry<T, Value>;
+// TODO(ECR-3979): replace with Entry
+type Index<T> = ProofEntry<T, Value>;
```

Comment:
I decided to change it to ProofEntry temporarily, because index hash will be unavailable otherwise.
Side note: do we ever need to implement support for Entry?

Reply:
The thing is, Java Entry does not have object_hash yet:

https://github.com/exonum/exonum-java-binding/blob/e096edce28f8aedc66487d9e4ed0fa1ec6c6f2aa/exonum-java-binding/core/src/main/java/com/exonum/binding/core/storage/indices/EntryIndexProxy.java#L173-L174

Reply:
🤔 Should I remove native getObjectHash then? It won't work with Entry

Reply:
I see, @slowli do you think we shall provide both Entry & ProofEntry? 

If yes, I'd remove the method, and make all the needed Java modifications separately to avoid confusion (Java needs the op, tests, documentation).

Reply:
The method impl seems to match the template for such methods and will be easy to copy-paste from an existing Proof index.

Reply:
Really funny, I wasn't able to change it back to `Entry` because than `getConsensusConfiguration` doesn't work - Core uses `ProofEntry` there. So I think it's better to have ProofEntry there before we implement it properly.

Reply:
OK, but let's rename it then — EntryIndexProxy -> ProofEntryIndexProxy (same with its ITs).

Reply:
> do you think we shall provide both Entry & ProofEntry?

These types both make sense, IMO. `ProofEntry` imposes certain overhead to compute the hash code of the contents each time it is set, while `Entry` does not. Also, since `Entry` is not aggregated into the state, it could be used for private data.

Reply:
OK, thanks, I think then we shall replace the present Entry with a ProofEntry (required for CoreSchema), and might add Entry as an improvement later (it won't be a breaky).